### PR TITLE
Add repo visibility to migrate-repo commands

### PR DESCRIPTION
**Pull request by tambling**
This PR adds the first half of https://github.com/github/gh-gei/issues/5: adding `--target-repo-visibility` to our `migrate-repo` commands, which will then send that setting up to the monolith's `StartRepositoryMigration` mutation (which handles the rest).

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [ ] Docs updated (or issue created)

### CODE REVIEW

File: src/ado2gh/Commands/MigrateRepoCommand.cs [hunk 1]
```diff
@@ -41,6 +42,11 @@ public class MigrateRepoCommand : Command
             {
                 IsRequired = true
             };
+            var targetRepoVisibility = new Option<string>("--target-repo-visibility")
```
Comment (tambling) [inline src/ado2gh/Commands/MigrateRepoCommand.cs:45]: I'm of two minds about the naming of this: on the one hand, I like staying consistent, on the other hand, only `gei` really cares about `source` vs. `target` repo, and we can just call this `--github-repo-visibility` here.

Comment (dylan-smith) [reply src/ado2gh/Commands/MigrateRepoCommand.cs:45]: I could go either way, @timrogers or @roferg any opinion? when we add `--target-repo-visibility` to the 3 different migrate-repo commands, should we be consistent with `--target-repo-visibility` as the arg name for all 3, or should we do `gh gei migrate-repo --target-repo-visibility` 'gh ado2gh migrate-repo --github-repo-visibility` `gh bbs2gh migrate-repo --github-repo-visibility`

Comment (roferg) [reply src/ado2gh/Commands/MigrateRepoCommand.cs:36]: We should be consistent and stick with the `target` branding. So I'd use `--target-repo-visibility`.

Comment (dylan-smith) [reply src/ado2gh/Commands/MigrateRepoCommand.cs:36]: I'm cool with that, but to provide a bit more context, we already differ argument names between the 3 extensions. For example:
`ado2gh migrate-repo --ado-repo --github-repo`
`gei migrate-repo --source-repo --target-repo`
`bbs2gh migrate-repo --bbs-repo --github-repo`

Comment (dylan-smith) [reply src/ado2gh/Commands/MigrateRepoCommand.cs:36]: So Option 1 would be:
`ado2gh migrate-repo --ado-repo foo --github-repo foo --target-repo-visibility internal`
`gei migrate-repo --source-repo foo --target-repo foo --target-repo-visibility internal`
`bbs2gh migrate-repo --bbs-repo foo --github-repo foo --target-repo-visibility internal`

Option 2 would be:
`ado2gh migrate-repo --ado-repo foo --github-repo foo --github-repo-visibility internal`
`gei migrate-repo --source-repo foo --target-repo foo --target-repo-visibility internal`
`bbs2gh migrate-repo --bbs-repo foo --github-repo foo --github-repo-visibility internal`

Comment (roferg) [reply src/ado2gh/Commands/MigrateRepoCommand.cs:36]: Did we change to GitHub repo recently? Our docs for ADO2GH use `--target-repo`. My opinion would be to align uniformly across all of the extensions unless there is a unique situation that calls for difference. In this case that would be using `target`.

Comment (dylan-smith) [reply src/ado2gh/Commands/MigrateRepoCommand.cs:36]: that's an oversight docs are currently incorrect for ado2gh

File: RELEASENOTES.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
 - 6 months ago in [v0.10](https://github.com/github/gh-gei/releases/tag/v0.10), we deprecated the --ssh flag, making it silently do nothing. Now, we’ve removed the flag entirely, so the CLI will error if you try to specify it. If you have any scripts that include the --ssh flag, you must remove that flag or the script will break.
 - If `create-team` fails when linking an IdP group with an HTTP 400, we will retry
-- If `create-team` fails when removing the initial team member, it will retry
\ No newline at end of file
+- If `create-team` fails when removing the initial team member, it will retry
```
Comment (dylan-smith) [inline RELEASENOTES.md:3]: just testing something

**Comment by dylan-smith** [comment]
Closing this PR as we've redone the changes in #953

**Pull request closed without merging**